### ISSUE TITLE
[JIT][PGO] Incorrect weights produced by expand-casts optimizations

### ISSUE THREAD

**Issue by EgorBo** [dotnet/runtime#50030]
```csharp
static string Foo(object o) => (string)o;
```
Currently produces:
```asm
; Method Program:Foo(System.Object):System.String
G_M28009_IG01:              ;; offset=0000H
       4883EC28             sub      rsp, 40
						;; bbWeight=1    PerfScore 0.25

G_M28009_IG02:              ;; offset=0004H
       48894C2430           mov      gword ptr [rsp+30H], rcx
       488BC1               mov      rax, rcx
       4885C0               test     rax, rax
       740F                 je       SHORT G_M28009_IG04
						;; bbWeight=1    PerfScore 2.50

G_M28009_IG03:              ;; offset=0011H
       48BAB873D0E9F97F0000 mov      rdx, 0x7FF9E9D073B8
       483910               cmp      qword ptr [rax], rdx
       7505                 jne      SHORT G_M28009_IG05
						;; bbWeight=0.25 PerfScore 0.81

G_M28009_IG04:              ;; offset=0020H
       4883C428             add      rsp, 40
       C3                   ret      
						;; bbWeight=1    PerfScore 1.25

G_M28009_IG05:              ;; offset=0025H
       488BCA               mov      rcx, rdx
       488B542430           mov      rdx, gword ptr [rsp+30H]
       E83E7BACFF           call     CORINFO_HELP_CHKCASTCLASS_SPECIAL
       CC                   int3     
						;; bbWeight=0    PerfScore 0.00
; Total bytes of code: 51
```

Note that 0.25 weight which looks incorrect.

Same for some other cast patterns.

category:correctness
theme:profile-feedback
skill-level:intermediate
cost:small
impact:small

**Comment by EgorBo** [comment]
Even weirder weights here:
```csharp
static bool Foo(object o) => o is string;
```
```asm
; Method Program:Foo(System.Object):bool
G_M5401_IG01:              ;; offset=0000H
						;; bbWeight=1    PerfScore 0.00

G_M5401_IG02:              ;; offset=0000H
       4885C9               test     rcx, rcx
       7411                 je       SHORT G_M5401_IG05
						;; bbWeight=1    PerfScore 1.25

G_M5401_IG03:              ;; offset=0005H
       48B8B873CEE9F97F0000 mov      rax, 0x7FF9E9CE73B8
       483901               cmp      qword ptr [rcx], rax
       7402                 je       SHORT G_M5401_IG05
						;; bbWeight=0.25 PerfScore 0.81

G_M5401_IG04:              ;; offset=0014H
       33C9                 xor      rcx, rcx
						;; bbWeight=0.13 PerfScore 0.03

G_M5401_IG05:              ;; offset=0016H
       4885C9               test     rcx, rcx
       0F95C0               setne    al
       0FB6C0               movzx    rax, al
						;; bbWeight=1    PerfScore 1.50

G_M5401_IG06:              ;; offset=001FH
       C3                   ret      
						;; bbWeight=1    PerfScore 1.00
; Total bytes of code: 32
```

**Comment by EgorBo** [comment]
Fixed by the "late cast expansion" work in 9.0
